### PR TITLE
Add apptainer

### PR DESCRIPTION
**Pull request by multimeric**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by multimeric** [comment]
Hi @chrisburr, just pinging you as you asked. I wasn't able to get the `--version` working correctly (it reports the git commit instead of the version). Also none of the patches worked, so I've disabled all of them.

**Comment by chrisburr** [comment]
Have you tested this locally? The test's don't really work inside docker so the CI isn't able to validate that the binaries work properly.

File: recipes/apptainer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  description: |
+    Apptainer/Singularity is the most widely used container system for HPC. It is designed to execute applications at bare-metal performance while being secure, portable, and 100% reproducible. Apptainer is an open-source project with a friendly community of developers and users. The user base continues to expand, with Apptainer/Singularity now used across industry and academia in many areas of work.
+
+  dev_url: https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md
+  doc_url: https://apptainer.org/docs/
+
+extra:
+  recipe-maintainers:
+    - multimeric
```
Comment (chrisburr) [inline recipes/apptainer/meta.yaml:74]:
```suggestion
    - chrisburr
    - multimeric
```

File: recipes/apptainer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  version: {{ version }}
+
+source:
+  - folder: src/github.com/apptainer/apptainer
+    url: https://github.com/apptainer/apptainer/releases/download/v{{ version }}/apptainer-{{ version }}.tar.gz
+    sha256: 1772f25c8770f976abb82c29053b77183930426a83eb3e43b9f8068c3e88ac30
+    patches:
+      #- patches/0003-Use-external-CNI.patch
+      #- patches/0004-Find-unsquashfs-from-PATH.diff
+      #- patches/set-version-from-env.patch
```
Comment (chrisburr) [inline recipes/apptainer/meta.yaml:14]: This patch will probably need to be rebased, the others should be checked for if they're still needed.

File: recipes/apptainer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    - libuuid  # [linux]
+    - libarchive
+    - libseccomp  # [linux]
+    - openssl
+  run:
+    # CNI version from singularity/Gopkg.lock
+    - cni-plugins >=0.7.4  # [linux]
+    - squashfs-tools  # [linux]
+  run_constrained:
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx]
```
Comment (chrisburr) [inline recipes/apptainer/meta.yaml:42]: This will only ever be linux-only so no need for this:
```suggestion
```

Comment (multimeric) [reply recipes/apptainer/meta.yaml:42]: True, I wonder why this recipe ever had osx stuff in it since singularity never ran on osx.

**Comment by multimeric** [comment]
> Have you tested this locally? The test's don't really work inside docker so the CI isn't able to validate that the binaries work properly.

Yes, the recipe builds locally on my Linux machine.

**Comment by multimeric** [comment]
Does this relate to the first patch, which I disabled?

**Comment by multimeric** [comment]
Thanks for fixing this up, I appreciate it.

**Pull request merged**